### ISSUE TITLE
invalid parameter 'net50' after >= or < in '== net50' while upgrading netcoreapp3.1 to net5.0

### ISSUE THREAD

**Issue by imranypatel** [fsprojects/FAKE#2564]
### Description

Error while targetting net5.0 framework.

### Repro steps

1. dotnet new -i Saturn.Template
2. md saturn2net5
3. dotnet new saturn -lang F#
4. Change TFM in saturn2net5.fsproj TargetFramework as: net5.0
5. Change TFM in paket.dependencies as: framework: net5.0
6. dotnet new tool-manifest --force
7. dotnet tool install paket
8. dotnet tool install fake-cli
9. dotnet tool install saturn.cli
10. dotnet paket update
11. dotnet fake build -t run

### Expected behavior

Build and run project successfully on net5.0

### Actual behavior

Producing error:
> There was a problem while setting up the environment:
> -> Error during parsing of 'D:\Work\iprnd\fs\ipstackNG21\saturn2net5\paket.lock'.
> -> invalid parameter 'net50' after >= or < in '== net50'
> Hint: If you just upgraded the fake-runner you can try to remove the .fake directory and try again.

### Known workarounds

Still searching...

### Related information 

* Operating system Windows 10
*  Version of .Net Core
> .NET SDK (reflecting any global.json):
>  Version:   5.0.101
>  Commit:    d05174dc5a
> 
> Runtime Environment:
>  OS Name:     Windows
>  OS Version:  10.0.19041
>  OS Platform: Windows
>  RID:         win10-x64
>  Base Path:   C:\Program Files\dotnet\sdk\5.0.101\

* Version of Paket (5.257.0)
* Version of FAKE-CLI (5.20.3)
* Version of SATURN.CLI (0.12.1)
* Version of F#
> D:\Work\iprnd\fs\ipstackNG21\saturn2net5>dotnet fsi
> Microsoft (R) F# Interactive version 11.0.0.0 for F# 5.0

**Comment by halcwb** [comment]
I think this is the [same issue](https://github.com/fsharp/FAKE/issues/2565) as I have.

**Comment by seanamosw** [comment]
Using the pre-release 5.20.4-alpha.1642 works for me for now.

**Comment by halcwb** [comment]
Unfortunately not for me. And it's a pity that none of the maintainers are responding to these issues.

**Comment by matthid** [comment]
@halcwb Sorry, I kind of lost motivation for various reasons. Nobody volunteered yet to take over. And to be honest some people prefer the "the build as a project && dotnet run" approach which doesn't have issues like these and might be a way out of this as I guess you are not volunteering to take over.

**Comment by halcwb** [comment]
@matthid. I am sorry to hear this. I do sympathise with your predicament. This is something that happens more often in other really wonderful F# open source projects. It's the burden of maintenance. Unfortunately, I am not really qualified to help you, I am a full time pediatric intensive care doctor who can only work on software projects in spare time. 

Doesn't Microsoft or other companies invest any resources to help you? It would be a shame if Fake would just wither away like this.

**Comment by Thorium** [comment]
I get this error without targeting .net50
Paket version 5.257.0
My paket references says `frameworks: net461, netcoreapp3.1, netstandard2.0`
...but Paket has itself put net50 to paket.lock in some reference components.

**Comment by halcwb** [comment]
> @halcwb Sorry, I kind of lost motivation for various reasons. Nobody volunteered yet to take over. And to be honest some people prefer the "the build as a project && dotnet run" approach which doesn't have issues like these and might be a way out of this as I guess you are not volunteering to take over.

@matthid I have 'upgraded' your issue to the [F# forum](https://forums.fsharp.org/t/can-f-afford-losing-fake-and-other-libraries/1642). Hope you agree, but I think this is quite relevant in general.

**Comment by Thorium** [comment]
If I manually replace from paket.lock
- `(< net50)`
- `- restriction: >= net50`
- `(>= net50)`

...with empty string, then I'm able to run the build.

**Comment by Thorium** [comment]
How is non-merged PR-fix not any action?

**Comment by ScottShingler** [comment]
This issue has not been resolved and should be reopened.

**Comment by dsyme** [comment]
@yazeedobaid Note there are many issues which were automatically closed by the auto-closing agent bot which was set up in this repo (I don't know why...)

**Comment by yazeedobaid** [comment]
> @yazeedobaid Note there are many issues which were automatically closed by the auto-closing agent bot which was set up in this repo (I don't know why...)

@dsyme Thanks for the note. I have disabled the stable bot in the .NET 6 support PR, in [this commit](https://github.com/fsprojects/FAKE/pull/2609/commits/6a6e44a3c9f1c43ea591fb3af76dbd5ea170acfe)
Thanks

**Comment by CumpsD** [comment]
I encountered this as well today, here's a twitter convo with @forki indicating Paket supports it but the Paket in FAKE is probably outdated: https://twitter.com/sforkmann/status/1458104833723060226

**Comment by dsyme** [comment]
We do need an update to FAKE fairly soon that either takes out the Paket support, or makes it optional, or brings it up-to-date.  FAKE is currently unusable as soon as Paket is used with anything net60.

**Comment by CumpsD** [comment]
What about pushing this to nuget? https://github.com/fsprojects/FAKE/blob/release/next/RELEASE_NOTES.md#5205-alpha---2021-11-07

**Comment by dsyme** [comment]
Yes I think we should be pushing it to nuget ASAP - even if it's marked pre-release.  @yazeedobaid is looking at the release pipeline I think

**Comment by atlemann** [comment]
Replacing

```csharp
#r "paket: groupref Build //"`
```

with the following

```csharp
#r "nuget:Fake.Core.Process"
#r "nuget:Fake.Core.Environment"
#r "nuget:Fake.Core.SemVer"
#r "nuget:Fake.Core.Target"
#r "nuget:Fake.DotNet.Cli"
#r "nuget:Fake.DotNet.MSBuild"
#r "nuget:System.Reactive"
#r "nuget:MSBuild.StructuredLogger,2.1.507" // Workaround for issue https://github.com/fsprojects/FAKE/issues/2595

let execContext = Fake.Core.Context.FakeExecutionContext.Create true "" []
Fake.Core.Context.setExecutionContext (Fake.Core.Context.RuntimeContext.Fake execContext)
```

and running with

```bash
dotnet fsi build.fsx
```

might be a workaround

**Comment by voronoipotato** [comment]
if https://github.com/fsharp/fslang-suggestions/issues/1144 were implemented using fsi might be a viable solution rather than just a workaround.

**Comment by Thorium** [comment]
This will probably hit after every new .NET major version release.

**Comment by Thorium** [comment]
Paket dependency has been updated, helps this. For future issues: Paket dependency has to be kept up-to-date.
